consensus/istanbul/core/handler.go

```diff
@@ -206,11 +231,10 @@ func (c *core) handleTimeoutMsg() {
 		}
 	}
 
-	lastProposal, _ := c.backend.LastProposal()
-	if lastProposal != nil && lastProposal.Number().Cmp(c.current.Sequence()) >= 0 {
+	if lastProposal.Number().Cmp(c.current.Sequence()) >= 0 {
 		c.logger.Trace("round change timeout, catch up latest sequence", "number", lastProposal.Number().Uint64())
 		c.startNewRound(common.Big0)
 	} else {
-		c.sendNextRoundChange("handleTimeoutMsg. lastProposal is nil or lastProposal's number is smaller than current sequence")
+		c.sendRoundChange(view.Round)
```

Comment:
What do you think about replacing all `sendNextRoundChange` calls to `sendRoundChange`? `sendNextRoundChange` is still called in `core.commit()' and  `core.handlePreprepare()`. Using `sendNextRoundChange` may cause the same problem that is fixed here.

Reply:
The round change timer is the only routine related to `round` executing concurrently. After this change, other routines calling `sendNextRoundChange` will not have a concurrent issue since they doesn't access some values concurrently. 
Anyway, removing `sendNextRoundChange` sounds good for me. I will consider it later.